### PR TITLE
HADOOP-18823. Add Labeler Github Action.

### PR DESCRIPTION
**Pull request by ayushtkn**
### Description of PR

Add Labeler Github Action
https://github.com/marketplace/actions/labeler

### How was this patch tested?

My Fork: https://github.com/ayushtkn/hadoop/pulls

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  yamllint  |   0m  0s |  |  yamllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  30m  6s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shadedclient  |  20m 33s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   |  54m  5s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5874/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5874 |
| Optional Tests | dupname asflicense codespell detsecrets yamllint |
| uname | Linux 182028e78cdb 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d4c55e62e0d1d80edabe0a9dd7e4a4e5c28a65ad |
| Max. process+thread count | 554 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5874/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ayushtkn** [comment]
Result can be checked from my fork:
https://github.com/ayushtkn/hadoop/pulls

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  - dev-support/**
+BUILD:
+  - any: [ '/**/pom.xml' ]
+COMMON:
+  - hadoop-common-project/**
+HDFS:
+  - hadoop-hdfs-project/**
+RBF:
+  - hadoop-hdfs-project/hadoop-hdfs-rbf/**
+HDFS-NATIVE:
```
Comment (Hexiaoqiao) [inline .github/labeler.yml]: Show `NATIVE` name include hdfs-native and common-native(hadoop-common-project/hadoop-common/src/main/native) together?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#
```
Comment (Hexiaoqiao) [inline .github/labeler.yml:1]: Blank here.

**Review by Hexiaoqiao — COMMENTED** [review]
Great feature. leave nit comment inline, not blocker issues.
+1 from my side.

File: .github/labeler.yml (hunk 2 above)
Comment (ayushtkn) [reply .github/labeler.yml:1]: Added

File: .github/labeler.yml (hunk 1 above)
Comment (ayushtkn) [reply .github/labeler.yml]: Done

**Comment by ayushtkn** [comment]
Thanx @Hexiaoqiao for the review, have addressed the comments.
Example PR from my fork: https://github.com/ayushtkn/hadoop/pull/8

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 27s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  yamllint  |   0m  0s |  |  yamllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  30m  5s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shadedclient  |  20m 32s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   |  54m 20s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5874/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5874 |
| Optional Tests | dupname asflicense codespell detsecrets yamllint |
| uname | Linux 4b41f8a5c46c 4.15.0-213-generic #224-Ubuntu SMP Mon Jun 19 13:30:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / cfc89d2ea5a768eaed9b986efc638b6d3c686f94 |
| Max. process+thread count | 697 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5874/5/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Hexiaoqiao — APPROVED** [review]
+1.

**Pull request merged**